### PR TITLE
Fix broken lists

### PR DESCRIPTION
**Pull request by cliftonc**
Addresses a bug with lists that are empty and cause the parser to crash.

Note:  I had an issue with types, and had to add a definition to fix it, if there is a better solution please let me know and I will fix it!

### CODE REVIEW

**Comment by cliftonc** [comment]
... and I've just seen that someone else has already fixed it but it isn't merged!  Should have checked the PRs first.  https://github.com/instantish/martian/pull/3

Happy to close in favour of that one.

File: src/@types/third-party.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+declare module 'storybook-router' {
+  global {
+    interface Window {
+      location: object;
+    }
+  }
+}
```
Comment (juliarosenson) [inline src/@types/third-party.d.ts:7]: Can we remove storybook from this project?

**Review by juliarosenson — COMMENTED** [review]
Thanks for contributing this fix! Would like to merge it, just without storybook

File: src/@types/third-party.d.ts (hunk 1 above)
Comment (cliftonc) [reply src/@types/third-party.d.ts:7]: I didn't add it, but without this type file I couldn't compile, so was unsure how to fix!

Comment (juliarosenson) [reply src/@types/third-party.d.ts:7]: What are you doing to compile/what error do you see? I am able to run `tsc` to compile

Comment (cliftonc) [reply src/@types/third-party.d.ts:7]: It happened when I first checked it out:

```
➜  martian git:(upgrade-fix) ✗ npm run compile

> @instantish/martian@1.0.3 compile
> tsc

../node_modules/@types/reach__router/index.d.ts:14:49 - error TS2304: Cannot find name 'Window'.

14 export type WindowLocation<S = LocationState> = Window['location'] & HLocation<S>;
                                                   ~~~~~~


Found 1 error.

npm ERR! code 1
npm ERR! path /Users/cliftoncunningham/work/martian
npm ERR! command failed
npm ERR! command sh -c tsc
```

Comment (juliarosenson) [reply src/@types/third-party.d.ts:7]: Hmm I cloned your fork, deleted the storybook references, and was able to compile. I notice the error is coming from `node_modules/@types/reach__router/` but when I install the dependencies for this project, that does not get added to my local node modules

Comment (cliftonc) [reply src/@types/third-party.d.ts:7]: Very strange.  I even tried deleting `node_modules`, installing a clean node (upgrading to 17), and re-installing.  It's still there on mine.  No idea what to do or if this impacts others?

Comment (cliftonc) [reply src/@types/third-party.d.ts:7]: Whats even crazier, is if I run `yarn why @types/reach__router` ... 

```
➜  martian git:(upgrade-fix) ✗ yarn why @types/reach__router
yarn why v1.22.5
[1/4] 🤔  Why do we have the module "@types/reach__router"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
error We couldn't find a match!
✨  Done in 0.36s.
```

Comment (cliftonc) [reply src/@types/third-party.d.ts:7]: When you say `deleted the storybook references` ... what do you mean?

Comment (juliarosenson) [reply src/@types/third-party.d.ts:7]: that is weird! I deleted the file src/@types/third-party.d.ts and the line `"src/@types/*",` from tsconfig.json

**Comment by juliarosenson** [comment]
Fixed by #5

**Pull request closed without merging**